Binary files librtmp-org/.DS_Store and librtmp/.DS_Store differ
diff -uNrp librtmp-org/Makefile librtmp/Makefile
--- librtmp-org/Makefile	2010-06-30 14:52:14.000000000 -0400
+++ librtmp/Makefile	2010-07-07 12:38:32.000000000 -0400
@@ -33,10 +33,12 @@ MANDIR=$(DESTDIR)$(mandir)
 
 LIBS_posix=
 LIBS_mingw=-lws2_32 -lwinmm -lgdi32
+LIBS_darwin=
 LIBS=$(CRYPTO_LIB) -lz $(LIBS_$(SYS)) $(XLIBS)
 
 THREADLIB_posix=-lpthread
 THREADLIB_mingw=
+THREADLIB_darwin=-lpthread
 THREADLIB=$(THREADLIB_$(SYS))
 SLIBS=$(THREADLIB) $(LIBS)
 
@@ -45,6 +47,7 @@ INCRTMP=librtmp/rtmp_sys.h librtmp/rtmp.
 
 EXT_posix=
 EXT_mingw=.exe
+EXT_darwin=
 EXT=$(EXT_$(SYS))
 
 all:	$(LIBRTMP) progs
diff -uNrp librtmp-org/librtmp/Makefile librtmp/librtmp/Makefile
--- librtmp-org/librtmp/Makefile	2010-06-30 16:01:39.000000000 -0400
+++ librtmp/librtmp/Makefile	2010-07-07 14:10:25.000000000 -0400
@@ -27,8 +27,15 @@ CRYPTO_DEF=$(DEF_$(CRYPTO))
 
 SO_posix=so.0
 SO_mingw=dll
+SO_darwin=so.0
 SO_EXT=$(SO_$(SYS))
 
+SO_LDFLAGS_posix=-shared -Wl,-soname,$@
+SO_LDFLAGS_mingw=
+SO_LDFLAGS_darwin=-bundle -flat_namespace -undefined suppress -fno-common \
+  -headerpad_max_install_names $(XLDFLAGS)
+SO_LDFLAGS=$(SO_LDFLAGS_$(SYS))
+
 SHARED=yes
 SODEF_yes=-fPIC
 SOLIB_yes=librtmp.$(SO_EXT)
@@ -61,7 +68,7 @@ librtmp.a: $(OBJS)
 	$(AR) rs $@ $?
 
 librtmp.$(SO_EXT): $(OBJS)
-	$(CC) -shared -Wl,-soname,$@ $(LDFLAGS) -o $@ $^ $> $(CRYPTO_LIB)
+	$(CC) $(SO_LDFLAGS) $(LDFLAGS) -o $@ $^ $> $(CRYPTO_LIB)
 	ln -sf $@ librtmp.so
 
 log.o: log.c log.h Makefile
